src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+                car = createCarWithoutDrivers(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car", throwable);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id, model, manufacturer_id, name, country FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id=m.id "
+                + "WHERE c.is_deleted=false;";
+        List<Car> cars;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            cars = new ArrayList<>();
+            while (resultSet.next()) {
+                Car car = createCarWithoutDrivers(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of cars", throwable);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    private Car createCarWithoutDrivers(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer =
+                new Manufacturer(manufacturerId, manufacturerName, manufacturerCountry);
+        Long carId = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        return new Car(carId, model, manufacturer, null);
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getDriversQuery = "SELECT id, name, license_number FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id=cd.driver_id "
+                + "WHERE is_deleted=false AND cd.car_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriverStatement =
+                        connection.prepareStatement(getDriversQuery)) {
+            getDriverStatement.setLong(1, carId);
+            ResultSet resultSet = getDriverStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(parseDriversFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of drivers", throwable);
+        }
+    }
+
+    private Driver parseDriversFromResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
по факту кидаєш одного водія
```suggestion
    private Driver parseDriverFromResultSet(ResultSet resultSet) throws SQLException {
```